src/averageAges.js

```diff
@@ -37,7 +48,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const checkForChild = people.filter((person) => withChildren
+    ? person.sex === 'f' && people.find(child => child.mother === person.name)
+    : person.sex === 'f');
```

Comment:
you dont use element, that FIND method returns, so you shoul use SOME method that returns just boolean value.

and if current person, that you are cheking has name, that some of any persons has in "mother" propperty, it means that this person is exactly female (so in this case sex === 'f' cheсking is redundant)

```suggestion 
    const checkForChild = people.filter((person, index, array) => withChildren
      ? array.some((otherPerson) => otherPerson.mother === person.name)
      : sex === 'f';
  });
```